sycl/source/detail/common.cpp

```diff
@@ -12,6 +12,11 @@
 namespace sycl {
 __SYCL_INLINE_VER_NAMESPACE(_V1) {
 namespace detail {
+/// @brief CodeLocation information slot in thread local storage
+/// @details This structure is maintained by the SYCL runtime to manage the
+/// propagation of the code_location data down the stack without breaking ABI
+/// compatibility
```

Comment:
I guess this is the reasoning then. Do we expect to eliminate this during next ABI breaking window?

Also, I'm still not sure how it would work for the old binaries running with the new RT. What code location will we have in the TLS? How is it going to be useful?

Reply:
SYCL will continue to grow over time and we will need to balance when to accommodate ABI breakage and when to provide end user notification. Code location information will make it complete, but with lack of visibility today, end users welcome notifications for missing functionality with partial information and augmenting them with code location like end user information when the ABI breakage window is available.